common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1044,7 +1044,10 @@ public boolean contentEquals(CharSequence a) {
         if (a instanceof AsciiString) {
             return equals(a);
         }
-
+        final String asString = this.string;
+        if (asString != null) {
+            return asString.contentEquals(a);
```

Comment:
This is of great help in case framework writers have a `String`-based API for the Netty http headers, allowing a bound-free check (vectorized too) to check for equality

Reply:
I have to better think about this; what if a user create a cached AsciiString value providing a non Latin String? I am opening a separate issue for this, but right now is not safe use String equality
because  https://github.com/netty/netty/blob/a5cc7a89ab411212b7fabb5a57ff765686771d59/common/src/main/java/io/netty/util/AsciiString.java#L1407 doesn't perform any validation/transformation to make the string content the same as AsciiString